pkg/api/resource/quantity.go

```diff
@@ -372,41 +385,53 @@ func NewMilliQuantity(value int64, format Format) *Quantity {
 	}
 }
 
+// NewScaledQuantity returns a new Quantity representing the given
+// value * 10^scale in DecimalSI format.
+func NewScaledQuantity(value int64, scale Scale) *Quantity {
+	return &Quantity{
+		Amount: inf.NewDec(value, scale.infScale()),
+		Format: DecimalSI,
+	}
+}
+
 // Value returns the value of q; any fractional part will be lost.
 func (q *Quantity) Value() int64 {
-	if q.Amount == nil {
-		return 0
-	}
-	tmp := &inf.Dec{}
-	return tmp.Round(q.Amount, 0, inf.RoundUp).UnscaledBig().Int64()
+	return q.ScaledValue(0)
 }
 
-// MilliValue returns the value of q * 1000; this could overflow an int64;
+// MilliValue returns the value of ceil(q * 1000); this could overflow an int64;
 // if that's a concern, call Value() first to verify the number is small enough.
 func (q *Quantity) MilliValue() int64 {
+	return q.ScaledValue(Milli)
+}
+
+// ScaledValue returns the value of ceil(q * 10^scale); this could overflow an int64.
+// To detect overflow, call Value() first and verify the expected magnitude.
```

Comment:
Value() is the unscaled (a.k.a. `Scale = 0`) quantity. I think it will only overflow for quantities > 9 exa-units. Is that worth documenting there?